src/printer.rs

```diff
@@ -336,8 +336,8 @@ impl Printer {
                 header_string.push_str(key.as_str());
             }
             header_string.push_str(": ");
-            match value.to_str() {
-                Ok(value) => header_string.push_str(value),
+            match String::from_utf8(value.as_bytes().to_vec()) {
```

Comment:
You can use https://doc.rust-lang.org/std/str/fn.from_utf8.html to go directly from `&[u8]` to `&str`